Automattic/simplenote-electron
# Add Evernote Import Parser

Building on top of #922 with an Evernote importer! This only works with Evernote `.enex` files, and _does not_ work in the browser unfortunately due to the large file sizes of the Evernote exports (they contain binary data, base 64 encoded).

Evernote note data is exported in a nasty HTML format, so I've done my best to try and strip out all of the markup while conserving newlines. It also imports the tags 🏷 

It makes use of the `sax` module to stream the file while parsing to conserve memory. I've tested this with ~100MB file and it works well, but I would imaging large Evernote exports could be in the gigabytes easily so we should test that.

The caller of the import can receive emits of the progress of the import or when it fails:
```
const evernoteImporter = new EvernoteImporter({ noteBucket, tagBucket });
evernoteImporter.on('status', (eventName, details) => {
    console.log(eventName, details); // e.g. 'progress', 42
});

evernoteImporter.importNotes('/Users/user/test/evernote.enex');
```

**To Test**
* Pass `importEvernotes` a file path to an `.enex` Evernote export (full path, like `/Users/user/Desktop...`), along with the noteBucket and tagBucket.
* The notes should import and sync to your account!

**ToDo**
- [x] Passing progress or error messages back to the caller
